grpc/src/compiler/csharp_generator.h error

Hello,
  1. When I install grpc, there is an error: 

    ./src/compiler/csharp_generator.h:42:43: fatal error: google/protobuf/stubs/strutil.h: No such file or directory

 I check the .h files under google/protobuf/stubs/, and there is only a .h file called 'stl_util.h'. I use ubuntu 12.04

  2. By the way, should googler write code like "using namespace std;" in "grpc/src/compiler/csharp_generator.h" ?

@jtattermusch

--- a/src/compiler/csharp_generator.cc
+++ b/src/compiler/csharp_generator.cc
@@ -33,6 +33,7 @@
 
 #include <cctype>
 #include <map>
+#include <sstream>
 #include <vector>
 
 #include "src/compiler/csharp_generator.h"
@@ -44,7 +45,6 @@
 using google::protobuf::compiler::csharp::GetFileNamespace;
 using google::protobuf::compiler::csharp::GetClassName;
 using google::protobuf::compiler::csharp::GetUmbrellaClassName;
-using google::protobuf::SimpleItoa;
 using grpc::protobuf::FileDescriptor;
 using grpc::protobuf::Descriptor;
 using grpc::protobuf::ServiceDescriptor;
@@ -228,11 +228,14 @@ void GenerateStaticMethodField(Printer* out, const MethodDescriptor *method) {
 }
 
 void GenerateServiceDescriptorProperty(Printer* out, const ServiceDescriptor *service) {
+  std::ostringstream index;
+  index << service->index();
   out->Print("// service descriptor\n");
   out->Print("public static global::Google.Protobuf.Reflection.ServiceDescriptor Descriptor\n");
   out->Print("{\n");
   out->Print("  get { return $umbrella$.Descriptor.Services[$index$]; }\n",
-             "umbrella", GetUmbrellaClassName(service->file()), "index", SimpleItoa(service->index()));
+             "umbrella", GetUmbrellaClassName(service->file()), "index",
+             index.str());
   out->Print("}\n");
   out->Print("\n");
 }

--- a/src/compiler/csharp_generator.h
+++ b/src/compiler/csharp_generator.h
@@ -36,10 +36,7 @@
 
 #include "src/compiler/config.h"
 
-using namespace std;
-
 #include <google/protobuf/compiler/csharp/csharp_names.h>
-#include <google/protobuf/stubs/strutil.h>
 
 namespace grpc_csharp_generator {
 

